src/app.ts

```diff
@@ -47,37 +46,52 @@ app.head('/', (req, res) => {
 	res.send(200);
 });
 
+// swagger
 app.use('/docs', swaggerUi.serve, swaggerUi.setup(swagger));
 
-app.use((err: HttpException, req: Request, res: Response, next: NextFunction) => {
-	// set locals, only providing error in development
-	res.locals.message = err.message || 'unknown error';
-	res.locals.error = req.app.get('NODE_ENV') !== 'production' ? err : {};
-	const status = err.status || 500;
-
-	// tslint:disable-next-line no-console
-	console.error(err);
-
-	// render the error page
-	res.status(status);
-	res.render('error');
-});
+// error handler (has to be the last middleware)
+app.use(errorHandler);
 
+// the mongodb connection
 const db = mongoose.connection;
 // tslint:disable-next-line: no-console
-db.on('error', console.error.bind(logger, 'connection error:'));
-const mongoHost = `mongodb://${process.env.MONGO_HOST || 'localhost'}/notification-service`;
-logger.info('mongo host', { mongoHost });
-mongoose.connect(mongoHost);
+db.on('error', console.error.bind(logger, '[critical] MongoDB connection error:'));
```

Comment:
does this bind console.error to logger.error or logger?

Reply:
I was actually also uncertain about this one (came from the original code). I changed it so it's using the default logger.